carapace-server/src/main/java/org/carapaceproxy/server/RequestHandler.java

```diff
@@ -211,39 +211,44 @@ public void start() {
             case ACME_CHALLENGE:
             case REDIRECT:
                 return;
-            case PROXY:
+            case PROXY: {
+                EndpointConnection connection;
                 try {
 //                    LOG.log(Level.SEVERE, "TIME"+TIME_TRACKER.incrementAndGet()+" start " + this + " thread " + Thread.currentThread().getName());
-                    EndpointConnection connection = connectionToClient.connectionsManager.getConnection(new EndpointKey(action.host, action.port));
-                    connectionToEndpoint.set(connection);
-                    connection.sendRequest(request, this);
+                    connection = connectionToClient.connectionsManager.getConnection(new EndpointKey(action.host, action.port));                    
                 } catch (EndpointNotAvailableException err) {
                     fireRequestFinished();
                     LOG.log(Level.INFO, "{0} error on endpoint {1}: {2}", new Object[]{this, action, err});
                     return;
                 }
+                connectionToEndpoint.set(connection);
+                connection.sendRequest(request, this);
                 return;
-            case CACHE:
-                try {
-//                    LOG.log(Level.SEVERE, "TIME"+TIME_TRACKER.incrementAndGet()+" startc " + this + " thread " + Thread.currentThread().getName());
-                    cacheSender = connectionToClient.cache.serveFromCache(this);
+            }
+            case CACHE: {
+                cacheSender = connectionToClient.cache.serveFromCache(this);
                     if (cacheSender != null) {
-                        return;
-                    }
-                    EndpointConnection connection = connectionToClient.connectionsManager.getConnection(new EndpointKey(action.host, action.port));
-                    connectionToEndpoint.set(connection);
-                    cacheReceiver = connectionToClient.cache.startCachingResponse(request);
-                    if (cacheReceiver != null) {
-                        // https://tools.ietf.org/html/rfc7234#section-4.3.4
-                        cleanRequestFromCacheValidators(request);
-                    }
-                    connection.sendRequest(request, this);
+                    return;
+                }
+                    EndpointConnection connection;
+                try {
+//                    LOG.log(Level.SEVERE, "TIME"+TIME_TRACKER.incrementAndGet()+" startc " + this + " thread " + Thread.currentThread().getName());                    
+                    connection = connectionToClient.connectionsManager.getConnection(new EndpointKey(
```

Comment:
I am cleaning up the try-catch block, EndpointNotAvailableException can be thrown only by "getConnection"